spec.html

```diff
@@ -34644,7 +34644,7 @@ <h1>Static Semantics: Early Errors</h1>
             It is a Syntax Error if _NcapturingParens_ &ge; 2<sup>32</sup> - 1.
           </li>
           <li>
-            It is a Syntax Error if |Pattern| contains multiple |GroupSpecifier|s whose enclosed |RegExpIdentifierName|s have the same CapturingGroupName.
+            It is a Syntax Error if |Pattern| contains multiple |GroupSpecifier|s whose |GroupName|s have the same CapturingGroupName.
```

Comment:
```suggestion
            It is a Syntax Error if |Pattern| contains two or more |GroupSpecifier|s for which CapturingGroupName of the contained |GroupName| is the same.
```

I'd prefer something like this. Alternatively, we could probably rely on the chain rule here

```suggestion
            It is a Syntax Error if |Pattern| contains two or more |GroupSpecifier|s for which CapturingGroupName of |GroupSpecifier| is the same.
```

Reply:
Note that a GroupSpecifier can also be empty, i.e. it doesn't necessarily contain a GroupName.

So it might be more accurate to rephrase the first suggestion as:
> ... Pattern contains two or more GroupSpecifiers that each contain a GroupName, where CapturingGroupName of those GroupNames is the same.

With the second suggestion, the reader is left wondering what the CapturingGroupName of an empty GroupSpecifier is. We could define it to be a particular value, but then we'd have to exclude that value from the rule in question (because it isn't a syntax error for a pattern to have multiple empty GroupSpecifiers).

Another approach would be to define an SDO/AO that gathered all the CapturingGroupNames in a Pattern, and then the rule could say that it's a syntax error if that list has any duplicates.

Reply:
As discussed in the editor call today, we could also solve this by making GroupName not match empty input. If you don't think we should wait for that change (or it ends up being impractical), I can make another suggestion here. I 
still don't feel comfortable with the wording as-is.

Reply:
I'm going to take a look at that change to the grammar.

Reply:
Okay, I've added 3 commits:
1. Change _GroupSpecifier_ from nillable to optional.
2. Extend `CapturingGroupName` to _GroupSpecifier_ via chain rule.
3. Reword the early error rule via the second suggestion above.

They're logically separate but probably squashable.